### PR TITLE
Dynamic mode options

### PR DESCRIPTION
**Pull request by ShiftyRail**
[system] [administration]

image

See picture above.

The most important option is **no stacking**.
If it's active, the following rulesets are mutually exclusive :
- Revolution (any kind)
- Blob (any kind)
- Nuke ops (any kind)
- Cult
- Malf

Fully tested & all

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
Making a really good PR but also delivering a change you want that would need people to vote for. No stacking should be off by default and then toggled on by default in a separate PR mostly to preserve atomicity

**Comment by ShiftyRail** [comment]
That's fairly true

File: code/modules/admin/admin.dm [hunk 1]
```diff
@@ -732,6 +731,37 @@ var/global/floorIsLava = 0
...
+		<h4>Common options</h4>
+		<i>All thse options can be changed midround.</i> <br/>
+		<a href='?src=\ref[src];force_extended=1'>Force extended</a> <br/>
+		<b>Description:</b> This will force the round to be extended. No rulesets will be drafted. This option is currently <b>[dynamic_forced_extended ? "on" : "off"]</b>. <br/>
+		<a href='?src=\ref[src];no_stacking=1'>No stacking</a> <br/>
+		<b>Description:</b> Unless the threat goes above 90, only one "round-ender" ruleset will be drafted. This option is currently <b>[dynamic_no_stacking ? "on" : "off"]</b>. <br/>
+		<a href='?src=\ref[src];classic_secret=1'>Classic secret mode</a> <br/>
+		<b>Description:</b> Only one roundstart ruleset will be drafted. Only traitors and minor roles will latespawn. This option is currently <b>[dynamic_classic_secret ? "on" : "off"]</b>. <br/>
+		<a href='?src=\ref[src];high_pop_limit=1'>High population limit</a> <br/>
+		<b>Description:</b>The threshold at which "high pipulation overide" will be in effect. The current value is <b>[dynamic_classic_secret ? "on" : "off"]</b>. <br/>
```
Comment (boy2mantwicethefam) [inline code/modules/admin/admin.dm]: override

File: code/modules/admin/admin.dm [hunk 2]
```diff
@@ -732,6 +731,37 @@ var/global/floorIsLava = 0
 	usr << browse(dat, "window=admin2;size=280x370")
 	return
 
+/datum/admins/proc/dynamic_mode_options(mob/user)
+	var/dat = {"
+		<center><B><h3>Dynamic Mode Options</h3></B></center><hr>
+		<br/> 
+		<h4>Common options</h4>
+		<i>All thse options can be changed midround.</i> <br/>
```
Comment (boy2mantwicethefam) [inline code/modules/admin/admin.dm]: these

File: code/modules/admin/admin.dm (hunk 1 above)
Comment (SirSkeleto) [reply code/modules/admin/admin.dm]: pipulation

**Comment by ShiftyRail** [comment]
Changed the no stacking default option for atomicity & all

**Comment by SonixApache** [comment]
I was wondering if such a control panel would ever show up. Really good shit. 👌👌👌👌👌👌

**Comment by Lokiamis** [comment]
pipulation

**Comment by oversane** [comment]
B-but no stacking means we will basically never get to see the unique interaction between nuke ops and a blob!

**Comment by ShiftyRail** [comment]
image

Made the interface a bit clearer, added some links for changing the parameters midround.
Now need to test if the features actually work

**Comment by ShiftyRail** [comment]
Added some procs to make testing of mutually exclusive roundstart easier (can comment them out if needed)
Fixed roundstart rulesets not being mutually exclusives with one another
Added a new variable to control (the threshold at which several round enders can happen ; the default is 90, which under normal distribution happens 2% of the rounds)

I spend a good 2 hours testing this so I'd say it's ready for review :tm:

**Comment by robosperg** [comment]
dangerously based PR

**Comment by DamianX** [comment]
Does this get reset every round? Persistence would be nice.

**Comment by ShiftyRail** [comment]
I'm not sure. If a badmin force extended for a bus but forgets to untick it after round end, it could be really frustrating in my opinion.

Once this technical PR is merged, I will make a PR to change the default values (no stacking to 1 by default).

**Comment by DeityLink** [comment]
I think having a stacking treshold like you did is a better alternative to outright preventing stacking.

Could you possibly add a forced starting Threat Level?

**Comment by ShiftyRail** [comment]
@DeityLink
That is the "rigged threat number" setting.

**Pull request merged**